stdlib/LinearAlgebra/test/generic.jl

```diff
@@ -14,12 +14,16 @@ struct Quaternion{T<:Real} <: Number
     v3::T
 end
 Quaternion(s::Real, v1::Real, v2::Real, v3::Real) = Quaternion(promote(s, v1, v2, v3)...)
+Base.convert(::Type{Quaternion{T}}, s::Real) where {T <: Real} =
+    Quaternion{T}(convert(T, s), zero(T), zero(T), zero(T))
```

Comment:
I had to add the definition of this `convert` and `+` below to do the test I mentioned in https://github.com/JuliaLang/julia/pull/29634#discussion_r235246958